pkg/instrumentation/kafka/kafka.go

```diff
@@ -176,6 +192,29 @@ func (i *FetchesRecordIter) Done() bool {
 	return isDone
 }
 
+// EachRecord calls underlying kgo.FetchPartition.EachRecord and traces the message.
+func (fp *FetchPartition) EachRecord(fn func(rec *kgo.Record)) {
+	fp.FetchPartition.EachRecord(func(rec *kgo.Record) {
+		span := fp.client.startConsumerSpan(fp.ctx, rec)
+
+		ctx := rec.Context
+		if ctx == nil {
+			ctx = context.Background()
+		}
+
+		rec.Context = tracer.ContextWithSpan(ctx, span)
+
+		fn(rec)
+	})
+}
+
+// ConsumeRecord finishes the span for a particular record.
+func (fp *FetchPartition) ConsumeRecord(rec *kgo.Record) {
```

Comment:
@laynax yeah, I just didn't want to bring the context of the `span` to the function name here

Reply:
@laynax Because we don't want to leak the fact that under the hood we are creating the span and we need to manage it. Maybe `ConsumeRecord` is not the best name, perhaps `MarkRecordAsConsumed` sounds better?